app/main.py

```diff
@@ -1 +1,47 @@
-#  write your code here
+from __future__ import annotations
+
+
+class Animal:
+
+    def __init__(self, name: str, appetite: int,
+                 is_hungry: bool = True) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        return 0
+
+
+class Cat(Animal):
+
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name=name, appetite=3, is_hungry=is_hungry)
+
+    def catch_mouse(self) -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name=name, appetite=7, is_hungry=is_hungry)
+
+    def bring_slippers(self) -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list[Animal]) -> int:
+    total_appetite = 0
+    for animal in animals:
+        if animal.is_hungry:
```

Comment:
unnecessary else

Reply:
I don't understand,  what do you mean? Quote from assignment:**Implement feed_animals function which takes a list of animals. It should feed passed animals and return a sum of food points that are needed to feed all hungry animals from this list.**

Reply:
<img width="479" alt="image" src="https://user-images.githubusercontent.com/72562232/203950083-8ae97173-0095-475e-bb76-ea7f985b2c58.png">
you already have checked this in the `feed()` method

Reply:
Understand, i love Python